src/launchpad/components/Console.js

```diff
@@ -13,44 +17,38 @@ const Console = React.createClass({
 
   displayName: "Console",
 
-  getInitialState: function() {
-    return {
-      expressions: []
-    };
-  },
-
-  evaluate(expression) {
-    this.props.client.clientCommands.evaluate(expression, {})
-      .then(
-        result => this.addExpression(expression, result),
-        err => console.warn("Error when evaluating expression", err)
-      );
-  },
-
-  addExpression(expression, result) {
-    this.setState(prevState => ({
-      expressions: [{
-        input: expression,
-        packet: result,
-      }, ...prevState.expressions]
-    }));
-  },
-
-  clearResultsList() {
-    this.setState(prevState => ({
-      expressions: []
-    }));
-  },
-
   render: function() {
+    let {
+      actions,
+      expressions,
+    } = this.props;
+
     return dom.main({},
       Header({
-        evaluate: this.evaluate,
-        clearResultsList: this.clearResultsList
+        evaluate: expression => actions.evaluateInput(
+          this.props.client.clientCommands.evaluate,
+          expression
+        ),
+        clearResultsList: actions.clearExpressions
       }),
-      ResultsList({expressions: this.state.expressions})
+      ResultsList({ expressions })
     );
   }
 });
 
-module.exports = Console;
+function mapStateToProps(state) {
+  return {
+    expressions: selectors.getExpressions(state)
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    actions: bindActionCreators(require("../actions"), dispatch)
+  };
```

Comment:
Sorry,  I was in the car so had to be vague. This is what we do in the debugger. I don't mind the actions field, just wanted to call it out

```js
function mapDispatchToProps(dispatch) {
  return bindActionCreators(require("../actions"), dispatch)
};
```